draft-ietf-teep-protocol.md

```diff
@@ -1392,6 +1389,18 @@ The cipher suites defined above do not do encryption at the TEEP layer, but
 permit encryption of the SUIT payload (e.g., using {{I-D.ietf-suit-firmware-encryption}}).
 See {{security}} for more discussion of specific payloads.
 
+For the initial QueryRequest message, unless the TAM has more specific knowledge about the TEEP Agent
+(e.g., if the QueryRequest is sent in response to some underlying transport message that contains a hint),
+the message does not use one of the above cipher suites but instead uses COSE_Sign with multiple signatures,
```

Comment:
Suggest to remove "does not use one of the above cipher suites but instead". Since it always contains a signature. The statement intends to say one or more than one signature. So consider a revision like the following:

the message SHOULD use COSE_Sign with multiple signatures,

Reply:
A single cipher suite can contain multiple signatures.  The two cipher suites in the doc today are:
teep-operation-sign1-es256 = [ cose-sign1, cose-alg-es256 ]
teep-operation-sign1-eddsa = [ cose-sign1, cose-alg-eddsa ]

and conceptually the TAM here uses [ cose-sign, cose-alg-es256, cose-alg-eddsa ]
as the ciphersuite, if those are the only two signature algorithms it supports.

Reply:
I see it now with this explanation but still feel confusing with the first statement. A request has supported_cipher_suites from TAM that will list two above:

[ 
  teep-operation-sign1-es256,
  teep-operation-sign1-eddsa
]

as a list. Then the signature data object for the entire message will be COSE_Sign1 for the former (known agent algorithm case), or COSE_Sign in general that will contain signature value for each algorithm listed in the supported cipher suites. The statement "does not use one of the above cipher suites but instead" is somehow confusing in reading. Would it be clearer to say:

"the message signature does not use COSE_Sign1 for a support algorithm by the TEEP Agent from the above cipher suites"?

Reply:
Updated, is it better now?

Reply:
yes :)